src/test/java/io/strimzi/kafka/bridge/facades/AdminClientFacade.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.kafka.bridge.facades;
+
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.AdminClientConfig;
+import org.apache.kafka.clients.admin.CreateTopicsResult;
+import org.apache.kafka.clients.admin.DeleteTopicsResult;
+import org.apache.kafka.clients.admin.NewTopic;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+
+import java.util.Collections;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+
+/**
+ * Class AdminClientFacade used for encapsulate complexity and asynchronous code of AdminClient.
+ */
+public class AdminClientFacade {
+
+    private static final Logger LOGGER = LogManager.getLogger(AdminClientFacade.class);
+    private static AdminClient adminClient;
+    private static AdminClientFacade adminClientFacade;
+
+    public static synchronized AdminClientFacade create(String bootstrapServer) {
+
+        if (adminClientFacade == null) {
+
+            Properties properties = new Properties();
+            properties.setProperty(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServer);
+            adminClient = AdminClient.create(properties);
+            adminClientFacade = new AdminClientFacade();
+        }
+        return adminClientFacade;
+    }
+
+    /**
+     * Method createAsyncTopic used for the race condition between in-memory kafka cluster and also encapsulate the get
+     *
+     * @param topicName         name of the topic
+     * @param partitions        number of partitions
+     * @param replicationFactor number of replication factor
+     */
+    public void createAsyncTopic(String topicName, int partitions, int replicationFactor) throws ExecutionException, InterruptedException {
+        CreateTopicsResult createTopicsResult = adminClient.createTopics(Collections.singletonList(new NewTopic(topicName, partitions, (short) replicationFactor)));
+        createTopicsResult.all().get();
```

Comment:
it's not "Async" anymore if you block with `get` on the future

Reply:
Yeah, that's correct replaced.